src/libostree/ostree-repo-checkout.c

```diff
@@ -613,9 +613,12 @@ checkout_one_file_at (OstreeRepo                        *repo,
     }
 
   const gboolean is_symlink = (g_file_info_get_file_type (source_info) == G_FILE_TYPE_SYMBOLIC_LINK);
+  const guint32 source_mode = g_file_info_get_attribute_uint32 (source_info, "unix::mode");
+  const gboolean is_unreadable = (!is_symlink && (source_mode & ~S_IFMT) == 0);
```

Comment:
Should we be more strict here and also mask out setuid/setgid bits? E.g.

```suggestion
  const gboolean is_unreadable = (!is_symlink && (source_mode & (S_IRWXU | S_IRWXG | S_IRWXO)) == 0);
```

Reply:
Or, how about just `& S_IRUSR` ?

Reply:
SGTM

Reply:
Updated with that.